modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/GetFormReportDataMVCResourceCommand.java

```diff
@@ -48,9 +53,30 @@ protected void doServeResource(
 			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
 		throws Exception {
 
+		long formInstanceId = ParamUtil.getLong(
+			resourceRequest, "formInstanceId");
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)resourceRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		if (!themeDisplay.isSignedIn() ||
+			!_ddmFormInstanceModelResourcePermission.contains(
+				GuestOrUserUtil.getPermissionChecker(), formInstanceId,
+				ActionKeys.VIEW)) {
+
+			JSONPortletResponseUtil.writeJSON(
+				resourceRequest, resourceResponse,
+				JSONUtil.put(
+					"errorMessage",
+					_language.get(
+						_portal.getHttpServletRequest(resourceRequest),
+						"your-request-failed-to-complete")));
+
+			return;
+		}
+
 		try {
-			long formInstanceId = ParamUtil.getLong(
-				resourceRequest, "formInstanceId");
+			_ddmFormInstanceService.getFormInstance(formInstanceId);
```

Comment:
I don't think we need this since we already verify the permissions before.

Reply:
Done